frontend/src/containers/SignupView.tsx

```diff
@@ -1,69 +1,101 @@
 import { Form, Input, Icon, Button } from 'antd';
 import { FormComponentProps } from "antd/lib/form";
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { connect } from 'react-redux';
 import { RouteComponentProps, NavLink } from 'react-router-dom';
 
 import { authSignup } from "actions/auth";
+import { AuthError } from "reducers/auth";
 import { State } from "reducers/index";
 
 interface StateToProps {
   loading: boolean;
-  error: Error | null;
+  error: AuthError | null;
 }
 
 interface DispatchToProps {
   authSignup: (username: string, email: string, password1: string, password2: string) => void;
 }
 
-type Props = RouteComponentProps & StateToProps & DispatchToProps & FormComponentProps;
+interface ErrorProps {
+  error: AuthError;
+};
 
+const RegistrationError = ({ error }: ErrorProps) => {
+  if (error.response.status !== 400) {
+    return <div>{error.message}</div>;
+  }
+  const parsedErrors = [];
+  let key = 0;
+  for (const errors of Object.values(error.response.data)) {
+    for (const error of errors) {
+      parsedErrors.push(<li key={key}>{error}</li>);
+      key++;
+    }
+  }
+  return (
+    <div>
+      <ul>
+        {parsedErrors}
+      </ul>
+    </div>
+  )
+}
 
+type Props = RouteComponentProps & StateToProps & DispatchToProps & FormComponentProps;
 
-class RegistrationForm extends React.Component<Props> {
-  state = {
-    confirmDirty: false,
-  };
+const RegistrationForm = (props: Props) => {
+  const [submitted, setSubmitted] = useState(false);
+  const { loading, error, history } = props;
 
-  handleSubmit = (e: React.FormEvent) => {
+  useEffect(() => {
+    if (loading) { return; }
+    if (error === null && submitted) {
+      history.push('/');
```

Comment:
It would be nice to make the registration/login/sessions compliant with [OWASP guidelines](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html) but I'm gonna ignore it for now (and create an issue)